### ISSUE TITLE
m4gbust regression logs

### ISSUE THREAD

**Issue by ajrhacker** [mamedev/mame#10113]
@MooglyGuy, for help tracking down the cause of the regression in `m4gbust`, I have attached two error logs, produced with LOG_ALL enabled in `machine/6840ptm.cpp` (but with all driver-side logging in `barcrest/mpu4.*` disabled). The first is from top-of-tree except with 3eb3713e876019b4824546efab462c69bc175e10 reverted; the second is without reverting that commit.

[m4gbust_logs.zip](https://github.com/mamedev/mame/files/9162623/m4gbust_logs.zip)

**Comment by MooglyGuy** [comment]
Thank you, I appreciate you taking the time to write up a more detailed report - the casual "this is wrong" followed by ghosting over the past few weeks has really gotten under my skin. Probably wasn't your intention, but it's how it came off.

The most likely scenario is that `m4gbust` is being affected by the same issue that Coverity flagged, which is that `msb` being unsigned at line 185 will result in an always-false comparison. Change it to `if (msb == 0xffff)` and I suspect things will be in better shape. Probably best if you or someone else with direct commit access makes the change, since it will avoid a bunch of needless screwing around with pull requests.

**Comment by ajrhacker** [comment]
Unfortunately that fix does nothing for `m4gbust`, since the offending timer isn't operating in the dual 8-bit mode.

**Comment by MooglyGuy** [comment]
Fair enough. Please enable all possible logging and post the relevant contents of the file in a ZIP. I'm willing to look into it this weekend on my own, but I'll need more concrete info if it needs to be solved before then.
